### PR TITLE
ONXP-516 [Sprint 4] Complaints resolving

### PR DESCRIPTION
**Pull request by OlehHinko**

### CODE REVIEW

File: src/pages/admin-panel/requests/ComplaintsList.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+					</Button>
+				),
+			},
+			{
+				title: "Name performer",
+				render: res => (
+					<Button style={styles.btnColor} type="link" onClick={() => this.showUserData(res.taker)}>
+						<span>{res.taker.first_name + " " + res.taker.last_name}</span>
+					</Button>
+				),
```
Comment (goonerDp) [inline src/pages/admin-panel/requests/ComplaintsList.js]: Please, make for users easer to understand that name fields are clickable. For example, change them to button links or style as links.

File: src/pages/admin-panel/requests/ComplaintsList.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+				<Table
+					columns={columns}
+					loading={loadingRequestComplaintsData}
+					rowKey={data => data.id}
+					dataSource={data}
+					className="ovf-auto"
+					onChange={this.handleTableChange}
+					pagination={pagination}
+				/>
+				{visible ? (
```
Comment (goonerDp) [inline src/pages/admin-panel/requests/ComplaintsList.js]: It's not clear what is visible!

File: src/api/admin/complaints.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+				...params,
+			},
+		});
+		return data;
+	} catch (error) {
+		console.log(error);
+		return handleReqError(error);
+	}
+};
+
```
Comment (goonerDp) [inline src/api/admin/complaints.js]: Make this API call universal for all operation-requests (for admin).
https://github.com/OnyxPay/onyxpay.co-backend#get-operation-requests-for-admin-url-apiv1adminoperation-requests-method-get

File: src/pages/admin-panel/requests/ComplaintsList.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+			requestId,
+			loadingSolve,
+			userId,
+			pagination,
+		} = this.state;
+		const columns = [
+			{
+				title: "Type request",
+				dataIndex: "type",
+				render: res => (res ? res : "n/a"),
```
Comment (goonerDp) [inline src/pages/admin-panel/requests/ComplaintsList.js]: What is "res"? Please name things more clearly, for example, "record"

File: src/pages/admin-panel/requests/ComplaintsList.js [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+					<>
+						<Button
+							type="primary"
+							onClick={() =>
+								this.handleComplainedRequests(res.request_id, "winnerClient", res.maker.id)
+							}
+							style={styles.btn}
+							loading={res.maker.id === userId && res.request_id === requestId && loadingSolve}
+						>
+							Winner {res.maker.first_name + " " + res.maker.last_name}
```
Comment (goonerDp) [inline src/pages/admin-panel/requests/ComplaintsList.js]: What if the name will be too long? It's better to name "winner initiator" or "winner performer"

File: src/components/modals/admin/ShowUserData.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+			render: res => (res ? res : "n/a"),
+		},
+		{
+			title: "Successful operations",
+			dataIndex: "successful_operations",
+		},
+		{
+			title: "Unsuccessful operations",
+			dataIndex: "unsuccessful_operations",
+		},
```
Comment (goonerDp) [inline src/components/modals/admin/ShowUserData.js]: Combine "Successful operations" and "Unsuccessful operations" titles in the table header as in example
image

File: src/components/modals/admin/ShowUserData.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { Modal, Table } from "antd";
+
+const ShowUserData = props => {
```
Comment (goonerDp) [inline src/components/modals/admin/ShowUserData.js]: Make this component universal for usage on the whole site.

File: src/components/modals/admin/ShowUserData.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+	const { data, visible, hideModal } = props;
+	const columns = [
+		{
+			title: "Country",
+			dataIndex: "country",
+			render: res => (res ? res : "n/a"),
```
Comment (goonerDp) [inline src/components/modals/admin/ShowUserData.js]: What is "res" ?

File: src/pages/admin-panel/requests/ComplaintsList.js [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+						<Button
+							type="primary"
+							loading={res.taker.id === userId && res.request_id === requestId && loadingSolve}
+							onClick={() =>
+								this.handleComplainedRequests(res.request_id, "winnerAgent", res.taker.id)
+							}
+							style={styles.btn}
+						>
+							Winner {res.taker.first_name + " " + res.taker.last_name}
+						</Button>
```
Comment (goonerDp) [inline src/pages/admin-panel/requests/ComplaintsList.js]: When one button has preloader, disable the another.

File: src/pages/admin-panel/requests/ResolvedComplaints.js [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+			},
+			{
+				title: "Amount",
+				dataIndex: "amount",
+				render: res => (res ? convertAmountToStr(res, 8) : "n/a"),
+			},
+			{
+				title: "Winner",
+				dataIndex: "complainWinner",
+				render: res => (res ? res : "n/a"),
```
Comment (goonerDp) [inline src/pages/admin-panel/requests/ResolvedComplaints.js]: "complainWinner" is an object, you should read first and last names from it.

File: src/api/admin/complaints.js (hunk 3 above)
Comment (OlehHinko) [reply src/api/admin/complaints.js]: @goonerDp  I haven't  access to this repo

**Pull request merged**